File: files/CH-097.R
library(tidyverse)
library(readxl)
library(zoo)

path = "files/CH-097 Linear Interpolation.xlsx"
input = read_excel(path, range = "B2:E5")
test = read_excel(path, range = "H2:K15")

years = tibble(Year = 2010:2022)

df = years %>%
  left_join(input, by = c("Year" = "Year")) %>%
  mutate(across(c("A", "B", "C"), ~ na.approx(.x, na.rm = FALSE))) %>%
  mutate(across(
    c("A", "B", "C"),
    ~ case_when(
      row_number() == 1 ~ 2 * lead(.x) - lead(lead(.x)),
      row_number() == n() ~ 2 * lag(.x) - lag(lag(.x)),
      TRUE ~ .x
    )
  )) %>%
  mutate(across(c("A", "B", "C"), round, 0))

identical(df, test) # TRUE

File: files/000-099/097/CH-097.py
import pandas as pd
import numpy as np

path = 'CH-097 Linear Interpolation.xlsx'
input = pd.read_excel(path, usecols = "B:E", skiprows = 1, nrows = 3)
test  = pd.read_excel(path, usecols = "H:K", skiprows = 1, nrows = 13) 
test.columns = ['Year', 'A', 'B', 'C']

years = pd.DataFrame({'Year': range(2010, 2023)})
r1 = years.merge(input, how='left', on='Year')

r1[['A', 'B', 'C']] = r1[['A', 'B', 'C']].interpolate().round().astype(int)

r1.iloc[[0, -1], 1:] = 2 * r1.iloc[[1, -2], 1:] - r1.iloc[[2, -3], 1:]
print(r1.equals(test)) # True